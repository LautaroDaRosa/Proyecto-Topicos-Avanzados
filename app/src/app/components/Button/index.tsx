import React from 'react';
import styled, { css } from 'styled-components';

const StButton = styled.button<Omit<ButtonProps, 'text'>>`
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  transition: 0.1s;

  ${props =>
    props.action === 'primary' &&
    css`
      border: 1.5px solid #fca408;
      background-color: white;
      color: #fca408;

      &:disabled {
        border-color: #c0bfcb;
        color: #c0bfcb;
      }

      &:hover:enabled {
        cursor: pointer;
        color: white;
        background-color: #fca408;
      }

      &:active:enabled {
        color: white;
        background-color: #b27508;
        border-color: #b27508;
      }
    `};
`;

type ButtonProps = {
  type?: 'button' | 'submit' | 'reset';
  text: string;
  action?: string;
  disabled?: boolean;
};

const Button = ({ type, text, action, disabled }: ButtonProps) => (
  <StButton type={type} action={action} disabled={disabled}>
    {text}
  </StButton>
);

export default Button;
