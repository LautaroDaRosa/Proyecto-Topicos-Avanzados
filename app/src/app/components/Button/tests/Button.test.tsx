import React from 'react';
import Button from '../index';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';

describe('Button', () => {
  it('renders correctly when action is primary', () => {
    const { container } = render(<Button text="hello" action="primary" />);

    expect(screen.getByText('hello')).not.toBeNull();
    expect(container.firstChild).toHaveStyleRule('background-color', 'white');
    // snapshot test
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly when action is not primary', () => {
    const { container } = render(<Button text="hello" />);

    expect(screen.getByText('hello')).not.toBeNull();
    expect(container.firstChild).toHaveStyleRule('background-color', undefined);
    // snapshot test
    expect(container.firstChild).toMatchSnapshot();
  });
});
