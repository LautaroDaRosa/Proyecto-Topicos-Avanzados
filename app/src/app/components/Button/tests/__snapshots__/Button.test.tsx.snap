// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly when action is not primary 1`] = `
.c0 {
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

<button
  class="c0"
>
  hello
</button>
`;

exports[`Button renders correctly when action is primary 1`] = `
.c0 {
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  -webkit-transition: 0.2s;
  transition: 0.2s;
  border: 1.5px solid #fca408;
  background-color: white;
  color: #fca408;
}

.c0:disabled {
  border-color: #c0bfcb;
  color: #c0bfcb;
}

.c0:hover:enabled {
  cursor: pointer;
  color: white;
  background-color: #fca408;
}

.c0:active:enabled {
  color: white;
  background-color: #c70548;
  border-color: #c70548;
}

<button
  action="primary"
  class="c0"
>
  hello
</button>
`;
